### PR TITLE
Add autostart support for linux

### PR DESCRIPTION
**Pull request by ilya-fedin**
Fixes https://github.com/telegramdesktop/tdesktop/issues/422 https://github.com/telegramdesktop/tdesktop/issues/6780

### CODE REVIEW

File: Telegram/SourceFiles/platform/linux/specific_linux.cpp [hunk 1]
```diff
@@ -38,6 +42,61 @@ using Platform::File::internal::EscapeShell;
...
+	_backgroundInterface
+		.call(str_const_toString(kRequestBackground), QString(), _options);
+}
+
+void SandboxAutostart::disableAutostart() {
+	_options["autostart"] = false;
+
+	_backgroundInterface
+		.call(str_const_toString(kRequestBackground), QString(), _options);
+}
```
Comment (23rd) [inline Telegram/SourceFiles/platform/linux/specific_linux.cpp]: I don't quite understand why there are two methods here and why they cannot be combined into a single one that accepts a `bool` argument.

File: Telegram/SourceFiles/platform/linux/specific_linux.cpp [hunk 2]
```diff
@@ -38,6 +42,61 @@ using Platform::File::internal::EscapeShell;
...
+
+SandboxAutostart::SandboxAutostart()
+: _backgroundInterface(
+		str_const_toString(kPortalService),
+		str_const_toString(kPortalObjectPath),
+		str_const_toString(kBackgroundInterface)) {
+	_options["reason"] = tr::lng_settings_auto_start(tr::now);
+	_options["commandline"] = QStringList()
+		<< cExeName()
+		<< qsl("-autostart");
```
Comment (23rd) [inline Telegram/SourceFiles/platform/linux/specific_linux.cpp]: Seems like can be simplified.
```
_options["commandline"] = QStringList({ cExeName(), qsl("-autostart") });
```

**Comment by 23rd** [comment]
Actually, it doesn't work for me. Maybe I'm doing something wrong?
Is `./configure.sh -D TDESKTOP_API_TEST=ON -D DESKTOP_APP_USE_PACKAGED=OFF` not enough?

File: Telegram/SourceFiles/platform/linux/specific_linux.cpp [hunk 3]
```diff
@@ -416,6 +438,32 @@ bool psShowOpenWithMenu(int x, int y, const QString &file) {
...
+	if (home.isEmpty() || cAlphaVersion() || cExeName().isEmpty())
+		return;
+
+	if (InSandbox()) {
+#ifndef TDESKTOP_DISABLE_DBUS_INTEGRATION
+		if (start) {
+			SandboxAutostart(true);
+		} else {
+			SandboxAutostart(false);
+		}
```
Comment (23rd) [inline Telegram/SourceFiles/platform/linux/specific_linux.cpp]: `SandboxAutostart(start);` ? =)

Comment (ilya-fedin) [reply Telegram/SourceFiles/platform/linux/specific_linux.cpp]: Oops, fixed

**Comment by 23rd** [comment]
Okay, it seems working now.

**Comment by ilya-fedin** [comment]
I suggest to wait

**Pull request merged**